server/src/main/java/org/opensearch/search/aggregations/bucket/terms/SignificantTextAggregatorFactory.java

```diff
@@ -148,7 +148,9 @@ protected Aggregator createInternal(
             : includeExclude.convertToStringFilter(DocValueFormat.RAW, maxRegexLength);
 
         MapStringTermsAggregator.CollectorSource collectorSource = new SignificantTextCollectorSource(
```

Comment:
Is this the only invocation of this constructor? If so, I'd remove the parameter and initialize the SourceLookup internally in `SignificantTextCollectorSource`.

Reply:
@andrross It looks like this is the only invocation of this constructor, I can make that change.